tests/unit/test_cache.py

```diff
@@ -0,0 +1,101 @@
+import sys
+import os
+import unittest
+from mock import patch, mock_open
+from ads import Service, ServiceSet, Cache, _load_spec_file
+
+class MockDevice():
+    """MockDevice to suppress stdout"""
+
+    @classmethod 
+    def flush(s): pass
+    
+    def write(self, s): pass
+
+class TestCache(unittest.TestCase):
```

Comment:
I'm happy to be convinced that I'm wrong, but I find these unit tests to be pretty weird and confusing, and they don't convince me that cacheing works properly. What do you think of moving some/most of these to being bash tests?